src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -0,0 +1,155 @@
+        super(nodeJSExec, nodeJSPort, nodeJSArgs, nodeJSEnvironment);
+        this.ipAddress = ipAddress;
+        this.nodeJSExec = nodeJSExec;
+        this.nodeJSPort = nodeJSPort;
+        this.nodeJSArgs = nodeJSArgs;
+        this.nodeJSEnvironment = nodeJSEnvironment;
+        this.timeToWaitForServerIsStarted = timeToWaitForServerIsStarted;
+        this.timeUnitToWaitForServerIsStarted = timeUnitToWaitForServerIsStarted;
+    }
+
+    /**
+     * @return The base URL for the managed appium server.
+     */
+    @Override
+    public URL getUrl() {
+        try {
+            return new URL(String.format(URL_MASK, ipAddress,
+                    nodeJSPort));
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public boolean isRunning() {
+        if (process == null)
+            return false;
+
+        try {
+            ping(500, TimeUnit.MILLISECONDS);
+            return true;
+        } catch (UrlChecker.TimeoutException e) {
+            return false;
+        }
+
+    }
+
+    private void ping(long time, TimeUnit timeUnit) throws UrlChecker.TimeoutException{
+        URL url = getUrl();
+        try {
+            URL status = new URL(url.toString() + "/status");
+            new UrlChecker().waitUntilAvailable(time, timeUnit, status);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("There is something wrong with the URL " + url.toString().toString() + "/status");
+        }
+    }
+
+    /**
+     * Starts the defined appium server
+     * @throws AppiumServerHasNotBeenStartedLocallyException If an error occurs while spawning the child process.
+     * @see #stop()
+     */
+    public synchronized void start() throws AppiumServerHasNotBeenStartedLocallyException {
+
+        if (isRunning())
+            return;
+
+        try {
+            process = new CommandLine(this.nodeJSExec.getCanonicalPath(), nodeJSArgs.toArray(new String[] {}));
+            process.setEnvironmentVariables(nodeJSEnvironment);
+            process.copyOutputTo(System.err);
+            process.executeAsync();
+            ping(timeToWaitForServerIsStarted, timeUnitToWaitForServerIsStarted);
+        } catch (UrlChecker.TimeoutException e) {
+            destroyProcess();
+            String msgTxt = "The node  hasn't been run! Arguments: " + nodeJSArgs.toString() + " " + "\n";
+            String processStream = process.getStdOut();
+            if (!StringUtils.isBlank(processStream))
+                msgTxt = msgTxt + "Process output: " + processStream + "\n";
+
+            throw new AppiumServerHasNotBeenStartedLocallyException(msgTxt,
+                    e);
+        }
+        catch (Throwable e){
+            destroyProcess();
+            throw new AppiumServerHasNotBeenStartedLocallyException("An unexpected throwable has been caught!",
```

Comment:
Do you think it'd be more concise like this?

```java
throw new ServerNotStartedException("The local appium server has not been started.");
```

The message `An unexpected throwable has been caught!` isn't very useful to an end user.